core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.spec.ts

```diff
@@ -115,6 +136,42 @@ describe('DotEmaComponent', () => {
             });
         });
 
+        it('should open a confirm dialog and save on confirm when the iframe sends a postmessage with the delete-contenlet action', () => {
+            spectator.detectChanges();
+
+            const confirmDialogOpen = jest.spyOn(confirmationService, 'confirm');
+            const saveMock = jest.spyOn(store, 'savePage');
+            const confirmDialog = spectator.query(byTestId('confirm-dialog'));
+
+            window.dispatchEvent(
+                new MessageEvent('message', {
+                    origin: 'http://localhost:3000',
+                    data: {
+                        action: 'delete-contentlet',
+                        payload: {
+                            pageID: '123',
+                            container: {
+                                identifier: '123',
+                                uuid: '123'
+                            },
+                            pageContainers: [],
+                            contentletId: '123'
+                        }
+                    }
+                })
+            );
+
+            spectator.detectChanges();
+
+            expect(confirmDialogOpen).toHaveBeenCalled();
+
+            confirmDialog
+                .querySelector('.p-confirm-dialog-accept')
+                .dispatchEvent(new Event('click')); // This is the internal button, coudln't find a better way to test it
```

Comment:
I don't think there is a better way... not sure if someone else have a better approach.